File: client/src/app/modules/miscellaneous/services/legal.form.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { LegalForm } from '../../miscellaneous/model/LegalForm';

@Injectable({
  providedIn: 'root'
})
export class LegalFormService extends AppRestService<LegalForm>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getLegalForms() {
    return this.getListCached(new HttpParams(), "legal-forms");
  }

  addOrUpdateLegalForm(legalForm: LegalForm) {
    this.clearListCache(new HttpParams(), "legal-forms");
    return this.addOrUpdate(new HttpParams(), "legal-form", legalForm, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/miscellaneous/service/LegalFormService.java
package com.jss.osiris.modules.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.miscellaneous.model.LegalForm;

public interface LegalFormService {
    public List<LegalForm> getLegalForms();

    public LegalForm getLegalForm(Integer id);
	
	 public LegalForm addOrUpdateLegalForm(LegalForm legalForm);
}
